backend/routes.js

```diff
@@ -781,69 +781,73 @@ route.POST('/kv/{contractID}/{key}', {
     return Boom.unauthorized(null, 'shelter')
   }
 
-  const existing = await sbp('chelonia.db/get', `_private_kv_${contractID}_${key}`)
-
-  // Some protection against accidental overwriting by implementing the if-match
-  // header
-  // If-Match contains a list of ETags or '*'
-  // If `If-Match` contains a known ETag, allow the request through, otherwise
-  // return 412 Precondition Failed.
-  // This is useful to clients to avoid accidentally overwriting existing data
-  // For example, client A and client B want to write to key 'K', which contains
-  // an array. Let's say that the array is originally empty (`[]`) and A and B
-  // want to append `A` and `B` to it, respectively. If both write at the same
-  // time, the following could happen:
-  // t = 0: A reads `K`, gets `[]`
-  // t = 1: B reads `K`, gets `[]`
-  // t = 2: A writes `['A']` to `K`
-  // t = 3: B writes `['B']` to `K` <-- ERROR: B should have written `['A', 'B']`
-  // To avoid this situation, A and B could use `If-Match`, which would have
-  // given B a 412 response
-  const expectedEtag = request.headers['if-match']
-  if (!expectedEtag) {
-    return Boom.badRequest('if-match is required')
-  }
-  // "Quote" string (to match ETag format)
-  const cid = existing ? createCID(existing, multicodes.RAW) : ''
+  // Use a queue to prevent race conditions (for example, writing to a contract
+  // that's being deleeted or updated)
+  return sbp('chelonia/queueInvocation', contractID, async () => {
```

Comment:
Should this change be ported to #2737?

Reply:
This is more likely to affect #2737 but it's still a separate bug.